drivers/at86rf2xx/at86rf2xx_rf_ops.c

```diff
@@ -0,0 +1,971 @@
+        at86rf2xx_reg_write(dev, AT86RF2XX_REG__PAN_ID_0, le_pan.u8[0]);
+        at86rf2xx_reg_write(dev, AT86RF2XX_REG__PAN_ID_1, le_pan.u8[1]);
+        DEBUG("PANID: %04x\n", *pan_id);
+    }
+        break;
+    case IEEE802154_AF_PAN_COORD:
+        mutex_unlock(&dev->lock);
+        return -ENOTSUP;
+    }
+
+    mutex_unlock(&dev->lock);
+    return 0;
+}
+
+static int _config_src_addr_match(ieee802154_dev_t *hal, ieee802154_src_match_t cmd, const void *value)
+{
+    at86rf2xx_t *dev = hal->priv;
+    int res;
+    mutex_lock(&dev->lock);
+    switch(cmd) {
+        case IEEE802154_SRC_MATCH_EN: {
+            const bool en = *((const bool*) value);
+            at86rf2xx_set_option(dev, AT86RF2XX_OPT_ACK_PENDING, en);
+            res = 0;
+            break;
+        }
+        default:
+            res = -ENOTSUP;
+    }
+    mutex_unlock(&dev->lock);
+    return res;
+}
+
+static int _set_frame_filter_mode(ieee802154_dev_t *hal, ieee802154_filter_mode_t mode)
+{
+    at86rf2xx_t *dev = hal->priv;
+    bool promisc = false;
+    switch(mode) {
+        case IEEE802154_FILTER_ACCEPT:
+            promisc = false;
+            break;
+        case IEEE802154_FILTER_PROMISC:
+            promisc = true;
+            break;
+        case IEEE802154_FILTER_ACK_ONLY:
+        case IEEE802154_FILTER_SNIFFER:
+            return -ENOTSUP;
+    }
+
+    DEBUG("[at86rf2xx] set promiscuous to %i\n", promisc);
+    mutex_lock(&dev->lock);
+    at86rf2xx_set_option(dev, AT86RF2XX_OPT_PROMISCUOUS, promisc);
+    mutex_unlock(&dev->lock);
+    return 0;
+}
+
+int at86rf2xx_init(at86rf2xx_t *dev, const at86rf2xx_params_t *params, ieee802154_dev_t *hal, ieee802154_sec_context_t *sec, void (*cb)(void*), void *ctx)
+{
+    uint8_t tmp;
+    (void) tmp;
+
+    (void) at86rf2xx_periph;
+    mutex_init(&dev->lock);
+    mutex_lock(&dev->lock);
+#if AT86RF2XX_IS_PERIPH
+    (void) params;
+    /* set all interrupts off */
+    at86rf2xx_reg_write(dev, AT86RF2XX_REG__IRQ_MASK, 0x00);
+    at86rf2xx_periph = hal;
+#else
+    /* initialize device descriptor */
+    dev->params = *params;
+#endif
+
+    if (!AT86RF2XX_IS_PERIPH) {
+        at86rf2xx_spi_init(dev, cb, ctx);
+    }
+
+    at86rf2xx_hardware_reset(dev);
+
+    /* test if the device is responding */
+    if (at86rf2xx_reg_read(dev, AT86RF2XX_REG__PART_NUM) != AT86RF2XX_PARTNUM) {
+        DEBUG("[at86rf2xx] error: unable to read correct part number\n");
+        mutex_unlock(&dev->lock);
+        return -ENOTSUP;
+    }
+
+    hal->driver = &at86rf2xx_ops;
+    hal->priv = dev;
+    (void) sec;
+#if IS_USED(MODULE_IEEE802154_SECURITY) && \
+    IS_USED(MODULE_AT86RF2XX_AES_SPI)
+    sec->dev.cipher_ops = &_at86rf2xx_cipher_ops;
+    sec->dev.ctx = dev;
+#endif
+
+    if (!IS_ACTIVE(AT86RF2XX_BASIC_MODE)) {
+        at86rf2xx_set_option(dev, AT86RF2XX_OPT_AUTOACK, true);
+        at86rf2xx_set_option(dev, AT86RF2XX_OPT_CSMA, true);
+    }
```

Comment:
Isn't that something controlled by the radio HAL as opposed to some device specific pseudo-module?

Reply:
Yes, but we currently assume the radio will begin with those features enabled if there's hardware acceleration.
We could although overload `CONFIG_IEEE8202154_DEFAULT_MAX_FRAME_RETRANS` and `CONFIG_IEEE802154_DEFAULT_CSMA_CA_RETRIES` to decide whether to activate them during initialization or not.

Reply:
I think it makes sense to use generic config when possible instead of doing driver specific things